### PR TITLE
[#174162289] Select local app versions

### PR DESCRIPTION
**Pull request by balanza**
This PR introduces changes that allow to execute local version of the apps by just editing `.env` values

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,55 @@
...
+By default each app is executed against the last version on the `master` branch of its public repo. By using a different build context, it's possible to run local versions of the app. 
+
+### Database and Storage
+Database and storage are shared among each application in the mock.
+
+| name | description |
+|-|-|-|
+| STORAGE_CONNECTION_STRING | Connection string for the Azure Storage used by this mock |
+| STORAGE_ACCOUNT_NAME | Account name for the Azure Storage used by this mock |
+| STORAGE_ACCOUNT_KEY | Account key for the Azure Storage used by this mock |
```
Comment (gunzip) [inline README.md:40]:
```suggestion
```

**Comment by gunzip** [comment]
that's right but another way that enables hot reload of source changes is to have a docker compose override that binds the workdir to a local directory

**Comment by balanza** [comment]
> that's right but another way that enables hot reload of source changes is to have a docker compose override that binds the workdir to a local directory

What do you mean?

**Comment by gunzip** [comment]
even if you can change the docker context, you have to rebuild the docker container for each change in your code if you want to test a specific function locally when developing new features. the best developer experience would be to have hot reload on source changes (tsc -w + restart function app) which is doable only bypassing docker (afaik)

**Comment by balanza** [comment]
> even if you can change the docker context, you have to rebuild the docker container for each change in your code if you want to test a specific function locally when developing new features. the best developer experience would be to have hot reload on source changes (tsc -w + restart function app) which is doable only bypassing docker (afaik)

Hot reloading is about two things:
1. to have the app started in watch mode
2. to have local code directory mounted as a volume for the relative container.

The latter can be achieved by just adding this to the app definition: 
```yaml
    volumes: 
      - ${FUNCTIONS_APP_CONTEXT}:/usr/src/app
```
(knowing the image copies code into `/usr/src/app`)

Use this when `FUNCTIONS_APP_CONTEXT=/your/local/path` and it works already. The problem is it makes the app crash in case you're fetching from remote, so volume must be mounted conditionally depending on the case. It didn't occur me yet a way to do that.

**Comment by gunzip** [comment]
> The latter can be achieved by just adding this to the app definition:

yes, that's exactly what I meant with "have a docker compose override that binds the workdir to a local directory" :)

**Pull request merged**